### PR TITLE
Display error message for non-JS MAP scenarios

### LINKED ISSUE #4760: Display Error for non-JS scenario on MAP pages
**Is your feature request related to a problem? Please describe.**
Currently when a user visits a MAP on Simorgh, when Javascript is disabled we don't display an error, we render a white space. This doesn't give any feedback to the user:
 
Screenshot 2019-11-22 at 17 10 28

Currently on Responsive news we give the user some feedback: 
Screenshot 2019-11-22 at 17 10 57

We need to ensure we give feedback to the user when the player is unable to load. 
We should use the error message in these designs: 
https://zpl.io/bLEP6xJ

Current the error message is displayed in English - we will need to translate it across languages but we will render English for now

**Describe the solution you'd like**

Given a user visits a MAP page
when javascript is disabled on their browser 
and there is a poster image available for the MAP
then we will display the poster image, a grey overlay and the text "To play this content, please enable JavaScript, or try a different browser"

Articles have implemented the same solution here: https://github.com/bbc/simorgh/issues/2368
**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Cypress tests for both Canonical and AMP
Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by rebeccamcginn**
Resolves https://github.com/bbc/simorgh/issues/4760

**Overall change:**
Added an error message with placeholder image (if provided) for when a user accesses a MAP/Live Radio page with JavaScript disabled.

**Code changes:**

- Created a NoJSMediaMessage component to contain the no-JS message and placeholder image within a  <noscript> tag
- Nested the NoJSMediaMessage component within the psammead-media-player/Canonical and psammead-media-player/Amp Components.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: packages/components/psammead-media-player/src/Canonical/__snapshots__/index.test.jsx.snap [hunk 1]
```diff
@@ -11,12 +11,15 @@ exports[`Media Player: Canonical should render an iframe 1`] = `
...
+<div>
+  <iframe
+    allow="autoplay; fullscreen"
+    allowfullscreen=""
+    class="c0"
+    scrolling="no"
+    src="https://foo.bar/iframe"
+    title="Media player"
+  />
+  <noscript />
```
Comment (karinathomasbbc) [inline packages/components/psammead-media-player/src/Canonical/__snapshots__/index.test.jsx.snap:23]: Does this mean that <noscript /> will always render?

File: packages/components/psammead-media-player/src/NoJSMediaMessage/index.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  }
+`;
+
+const NoJSMediaMessage = ({
+  service,
+  noJsMessage,
+  placeholderSrc,
+  placeholderSrcset,
+}) => (
+  <noscript>
```
Comment (karinathomasbbc) [inline packages/components/psammead-media-player/src/NoJSMediaMessage/index.jsx]: As I mentioned above - should we always render the `<noscript>` tag, or should we check if we have a message to display, and only then render `<noscript>`?

**Review by ryanmccombe — COMMENTED** [review]
This looks great 👍 

Two small design considerations:

It's a bit more versatile (and idiomatic / "html-like") to pass the content within the body of the element, rather than as a prop, eg:

```
<NoJSMediaMessage service={service}>
  Please enable JS
</NoJSMediaMessage>
```

Also, I'd recommend considering decoupling the "No JS" part from the "Media Message" part.

The messaging design is likely to be independently useful (eg, for any error message, not just no-js) but the inclusion of `<noscript>` makes it restrictive.

Removing the `<noscript>` would add almost no overhead for any consumers who want to use it for the no-js use case - they just wrap it in a `noscript` themselves:
```
<noscript>
  <MediaMessage>
    Please enable JS
  </MediaMessage>
</noscript>
```
But we could now also do things like this:
```
if (stream.expired) return (
  <MediaMessage>Live stream expired</MediaMessage>
)
```

Removing the `noscript` would also solve the storybook / unit testing question :)

File: packages/components/psammead-media-player/src/MediaMessage/index.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+const MediaMessage = ({
+  service,
+  noJsMessage,
+  placeholderSrc,
+  placeholderSrcset,
+}) => (
+  <StyledWrapper>
+    {placeholderSrc && (
+      <Image alt="" src={placeholderSrc} srcset={placeholderSrcset} />
```
Comment (ryanmccombe) [inline packages/components/psammead-media-player/src/MediaMessage/index.jsx]: Not sure what the a11y certeria are, but we may want this to be `aria-hidden`

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute

File: packages/components/psammead-media-player/src/MediaMessage/index.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+`;
+
+const Message = styled.strong`
+  display: block;
+  font-weight: normal;
+  bottom: 0;
+  position: absolute;
+  padding: ${GEL_SPACING};
+  border-bottom: 0.0625rem solid transparent;
+  @media screen and (-ms-high-contrast: active) {
```
Comment (ryanmccombe) [inline packages/components/psammead-media-player/src/MediaMessage/index.jsx:44]: Nice - I think this is the first time I've seen a PR catch that before it's found in a11y testing 😄

File: packages/components/psammead-media-player/src/Canonical/index.jsx [hunk 5]
```diff
@@ -17,29 +26,44 @@ const Canonical = ({ src, title, placeholderSrc }) => {
...
+          placeholderSrcset={placeholderSrcset}
+        />
+      </noscript>
+    </>
   );
 };
 
 Canonical.propTypes = {
   src: string.isRequired,
+  placeholderSrcset: string,
```
Comment (ryanmccombe) [inline packages/components/psammead-media-player/src/Canonical/index.jsx:56]: We'll likely want a README update if we're adding or changing props - https://www.npmjs.com/package/@bbc/psammead-media-player

File: package.json [hunk 6]
```diff
@@ -75,6 +75,7 @@
     "@storybook/addon-viewport": "^5.3.9",
     "@storybook/react": "^5.3.9",
     "@storybook/theming": "^5.3.9",
+    "@testing-library/jest-dom": "^5.0.2",
```
Comment (ryanmccombe) [inline package.json:78]: Also needs a new record in CHANGELOG.md

File: packages/components/psammead-media-player/src/Amp/index.test.jsx [hunk 7]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { render } from '@testing-library/react';
 import Amp from '.';
+import '@testing-library/jest-dom/extend-expect';
```
Comment (ryanmccombe) [inline packages/components/psammead-media-player/src/Amp/index.test.jsx:5]: Usually want library imports above of our own files - surprised eslint isn't whinging about this, guess the rule is only in Simorgh

File: packages/components/psammead-media-player/src/MediaMessage/index.jsx (hunk 3 above)
Comment (rebeccamcginn) [reply packages/components/psammead-media-player/src/MediaMessage/index.jsx]: Added :)

**Review by ryanmccombe — APPROVED** [review]
source

File: package.json (hunk 6 above)
Comment (rebeccamcginn) [reply package.json:78]: Done :)

File: packages/components/psammead-media-player/src/Canonical/index.jsx (hunk 5 above)
Comment (rebeccamcginn) [reply packages/components/psammead-media-player/src/Canonical/index.jsx:56]: Done :)

**Review by jroebu14 — COMMENTED** [review]
Nice!

**Review by ryanmccombe — APPROVED** [review]
👍

File: packages/components/psammead-media-player/README.md [hunk 8]
```diff
@@ -1,34 +1,38 @@
...
+| `title`                   | string  | Yes      | -           | `'Video player'`                                                                                                                            |
+| `service`                 | string  | Yes      | -           | `'news'`                                                                                                                                    |
+| `showPlaceholder`         | boolean | No       | `true`      | `false`                                                                                                                                     |
+| `placeholderSrc`          | string  | No       | `null`      | `'http://foobar.com/placeholder.png'`                                                                                                       |
+| `placeholderSrcset`       | string  | No       | `''`        | `'https://bbc.com/300/cat.jpg 300w, https://bbc.com/450/cat.jpg 450w, https://bbc.com/600/cat.jpg 600w'`                                    |
```
Comment (karinathomasbbc) [inline packages/components/psammead-media-player/README.md:31]: Is the updated default value of '' instead of null intentional?

**Review by karinathomasbbc — APPROVED** [review]

File: packages/components/psammead-media-player/src/MediaMessage/index.test.jsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  const { getByText } = render(
+    <MediaMessage
+      service="news"
+      noJsMessage="Please enable Javascript or try a different browser"
+      placeholderSrc="http://foo.bar/placeholder.png"
+      placeholderSrcset="http://foo.bar/placeholder.png"
+    />,
+  );
+  const mediaMessage = getByText(
+    'Please enable Javascript or try a different browser',
```
Comment (greenc05) [inline packages/components/psammead-media-player/src/MediaMessage/index.test.jsx:16]: What HTML element will the message be wrapped in @rebeccarfmcginnbbc Just checking if this also follows the screen reader UX? https://app.zeplin.io/project/5d4d77b1fcbfc79bf47e0294/screen/5e2b162c84cd309968 Also any plans for an accessibility swarm on this? Maybe it's been done? Just checking...

**Pull request merged**